waspc/src/Wasp/Lib.hs

```diff
@@ -25,11 +25,13 @@ import qualified Wasp.Util.IO as Util.IO
 
 type CompileError = String
 
+type CompileWarning = String
+
 compile ::
   Path' Abs (Dir WaspProjectDir) ->
   Path' Abs (Dir ProjectRootDir) ->
   CompileOptions ->
-  IO (Either CompileError ())
+  IO (Either CompileError (Maybe CompileWarning))
```

Comment:
If we are going to change generator to return errors + warnings, then we will also want to change this I guess hm, to return multiple errors and warnings.

I can also tell you for sure that in the future we will also want to make Analyzer return multiple errors, and probably warnings at some point, so we will need it for sure.

Reply:
Yeah, here though I am building a returned string that can handle multiple warnings. So from the caller perspective either it failed with some error(s), or succeeded with some possible warning(s). I think we are still safe here if we are ok with a string containing multiples already, even with the updated return type, but curious if you have other ideas. Basically, do callers of `Lib.compile` need to care what the actual errors/warnings are at the type level, or just that they exist to be displayed. To me it feels only success/failure matters to the caller here, but idk as I type it. :D

Reply:
I also don't know! But you will certainly want to be able to return multiple errors and multiple warnings (right now it is either one error or maybe one warning). They can all still be strings though, for now.

I guess I find it confusing to see `CompilerWarning` that can then be multiple warnings. I would at least make it `[CompileWarning]` and then consumer/CLI can choose how to print these (all together, maybe some spacing in between, or smth).

Reply:
Yeah, that makes sense. Before when I said it returned multiple out of Lib.compile I just meant it did the concat of them before, but I think allowing the caller to process them individually does make sense. 👍🏻

Reply:
Updated signature and behavior to simply return:

```haskell
  (generatorWarnings, generatorErrors) <- Generator.writeWebAppCode appSpec outDir
  return (map show generatorWarnings, map show generatorErrors)
```